library/java/org/chromium/net/impl/CronetBidirectionalState.java

```diff
@@ -0,0 +1,431 @@
+    int ON_HEADERS = 19;            // EM invoked the "onHeaders" callback - response body to come.
+    int ON_HEADERS_END_STREAM = 20; // EM invoked the "onHeaders" callback - no response body.
+    int ON_DATA = 21;            // EM invoked the "onData" callback - not last "onData" callback.
+    int ON_DATA_END_STREAM = 22; // EM invoked the "onData" callback - final "onData" callback.
+    int ON_COMPLETE = 23;        // EM invoked the "onComplete" callback.
+    int ON_CANCEL = 24;          // EM invoked the "onCancel" callback.
+    int ON_ERROR = 25;           // EM invoked the "onError" callback.
+  }
+
+  /**
+   * Enum of the Next Actions to be taken.
+   */
+  @IntDef({NextAction.CARRY_ON, NextAction.WRITE, NextAction.FLUSH_HEADERS, NextAction.SEND_DATA,
+           NextAction.READ, NextAction.RECORD_READ_BUFFER, NextAction.INVOKE_ON_READ_COMPLETED,
+           NextAction.INVOKE_ON_ERROR_RECEIVED, NextAction.CANCEL,
+           NextAction.INVOKE_ON_WRITE_COMPLETED_CALLBACK,
+           NextAction.INVOKE_ON_READ_COMPLETED_CALLBACK, NextAction.INVOKE_ON_SUCCEEDED,
+           NextAction.INVOKE_ON_FAILED, NextAction.INVOKE_ON_CANCELED,
+           NextAction.TAKE_NO_MORE_ACTIONS})
+  @Retention(RetentionPolicy.SOURCE)
+  @interface NextAction {
+    int CARRY_ON = 0;                 // Do nothing special at the moment - keep calm and carry on.
+    int WRITE = 1;                    // Add one more ByteBuffer to the pending queue.
+    int FLUSH_HEADERS = 2;            // Start sending request headers.
+    int SEND_DATA = 3;                // Send one ByteBuffer on the wire, if any.
+    int READ = 4;                     // Start reading the next chunk of the response body.
+    int RECORD_READ_BUFFER = 5;       // Just save the ReadBuffer - read will occur just after.
+    int INVOKE_ON_READ_COMPLETED = 6; // Initiate the completion of a read operation.
+    int INVOKE_ON_ERROR_RECEIVED = 7; // Initiate the completion of a network Error.
+    int CANCEL = 8;                   // Tell EM to cancel. Can be an user induced, or due to error.
+    int INVOKE_ON_WRITE_COMPLETED_CALLBACK = 9; // Tell the User that a write operation completed.
+    int INVOKE_ON_READ_COMPLETED_CALLBACK = 10; // Tell the User that a read operation completed.
+    int INVOKE_ON_SUCCEEDED = 11;  // Tell the User the stream was completed successfully.
+    int INVOKE_ON_FAILED = 12;     // Tell the User the stream completed in an error state.
+    int INVOKE_ON_CANCELED = 13;   // Tell the User the stream completed in a cancelled state.
+    int TAKE_NO_MORE_ACTIONS = 14; // The stream is already in error state - don't do anything else.
+  }
+
+  /**
+   * Bitmap used to express the global state of the BIDI Stream. Each bit represent one element of
+   * the global state.
+   */
+  @IntDef(flag = true, // Not used as an Enum, and this is not used as the argument of a "switch".
```

Comment:
nit: "This is not used as an Enum nor as the argument of a switch statement."

Reply:
Thanks! Done.